### DISCUSSION TITLE
New GOTEK PCB design

### DISCUSSION THREAD

**Discussion by tizzulo** [keirf/flashfloppy#493] [General]
took the FF hex fine, no compatibility issues noted so far 
![B1B884F7-A57B-4976-9D83-ED8F5E237C43](https://user-images.githubusercontent.com/82121331/122143070-30fc4500-ce40-11eb-8c8a-b40639b9eaf2.jpeg)

**Comment by keirf** [comment]
Thanks for posting. This is the promised PCB revision with a dedicated rotary port: see the 3x2 header labelled JK and J8. It isn't supported yet. I need to speak to the Gotek factory about exact assignment of those header pins.

**Reply by keirf** [reply]
To be clear: For now you will need to treat this like the previous 48-pin PCB and connect rotary at the debug/programming port at the rear.

**Reply by prosper00** [reply]
I jist got one of these pcb's.  I can trace out the connections on the 3x2 header if you want.  It looks like at least some go directly to some otherwise unused pins on the mcu.  Anyone got a datasheet/pinouts for the mcu?

**Comment by pilou75** [comment]
there's one more PCB design
mine have also the J8/JK 3x2 header, but only 2x2 header for near J7
it is different from the picture in the flash procedure (msx.org)

**Comment by pasmoi87** [comment]
Hi, I have this pcb version with an AT32F415. Rotary at the debug/programming does not work.

**Reply by pasmoi87** [reply]
SFRKC30.AT2

**Reply by keirf** [reply]
You must use the new 3x2 rotary header described here https://github.com/keirf/FlashFloppy/wiki/Hardware-Mods#kc30-rotary-header

**Reply by olDirdey** [reply]
Is it totally dead or is your button working but you cant use to navigate? If that´s the case than take a look how you wired your button. On most rotary encoders ground is common for the button and rotary function. If you connect it in the wrong direction then SW from PCB is connected to GND from rotary. Your button is working now cause it only needs two cables to be shorted together. But not the rotary function cause ground is missing. Just switch the cables. My Board is the SFRKC30AT3 that was delivered with 6 Pin rotary KC30. But I installed 4 Pin KC40. 

![gotek](https://user-images.githubusercontent.com/17804482/146645561-cdc80c9f-a763-4794-b779-809195f91b75.jpg)

**Comment by pasmoi87** [comment]
Thanks, bad kuck for me I test with 2 pins of JK. And this is JK and J8.
